src/main/java/seedu/address/logic/commands/EditListingCommand.java

```diff
@@ -0,0 +1,209 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AREA;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRICE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REGION;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.listing.Address;
+import seedu.address.model.listing.Area;
+import seedu.address.model.listing.Listing;
+import seedu.address.model.listing.Price;
+import seedu.address.model.listing.Region;
+import seedu.address.model.person.Name;
+
+/**
+ * Edits the details of an existing listing in the system.
+ */
+public class EditListingCommand extends Command {
+
+    public static final String COMMAND_WORD = "editListing";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the listing identified "
+            + "by the index number used in the displayed listing list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
```

Comment:
Currently the `Edit` commands use `Index` to identify which `listing` or `client` to edit while the `Delete` commands use `Name` to do so. Should we standardise this process?